markovbot/commands.py

```diff
@@ -44,3 +47,84 @@
     await markovbot.say(sentence)
 
 
+@markovbot.command(pass_context=True, help='Mock the last specified user message.')
+async def mock(context, user: str = None):
+    server_context = context.server_context
+
+    if user is None:
+        await markovbot.say('Please specify a user')
+        return
+
+    else:
+        targeted_user_id = str()
+
+        # Get member from server with selected username
+        for member in server_context.server.members:
+            print('Member Nick: {}'.format(member.nick))
+            if member.nick == user:
+                print('Member Nick: {}'.format(member.nick))
+                targeted_user_id = member.id
+                return
+
+        # Get messages from user in current channel
+        message_logs = get_logs_from_channel(context.message.channel)
+
+        logs_by_user = list(filter(
+            lambda message_log: message_log.author.id == targeted_user_id, message_logs))
+        print('Filtered messages!')
+        # Get latest message from user
+        logs_by_user.sort(
+            key=lambda message_log: message_log.timestamp, reverse=True)
+        sentence = logs_by_user[0]
+        print('Got Latest Message!')
+
+    await markovbot.say(sentence)
+
+
+@asyncio.coroutine
+def get_logs_from_channel(channel):
```

Comment:
I was trying to fix the issue with iterating through the logs_from object. Apparently the way that it is done in the docs doesn't work on my instance. I haven't figured out why yet, but that's the only wall I'm hitting right now.

Reply:
You also can't do a yield from inside of an async method.

Reply:
Oh im dumb. Use `await` instead of yield. Check the other command above. 

This is how I got the logs in another spot https://github.com/SheldonB/discord-markov-bot/blob/master/markovbot/server.py#L62

Reply:
yield is for generator functions. I mixed them up, although I think that was actually how async/await pattern was implemented before the syntax was added. The annotation bascually wrapped your function in a generator function, and yielded at the async function.